src/Moat.hs

```diff
@@ -955,21 +983,23 @@ mkProd o@Options {..} typName instTys ts = \case
     { constructorVariant = RecordConstructor fieldNames,
       ..
     } -> do
-      let fields = zipFields o fieldNames constructorFields
-      matchProxy =<< lift (structExp typName instTys dataInterfaces dataProtocols dataAnnotations fields ts makeBase)
+      fieldDocs <- lift $ mapM getDoc fieldNames
+      let fields = zipFields o fieldNames constructorFields fieldDocs
+      matchProxy =<< lift (structExp typName parentDoc instTys dataInterfaces dataProtocols dataAnnotations fields ts makeBase)
 
-zipFields :: Options -> [Name] -> [Type] -> [Exp]
+zipFields :: Options -> [Name] -> [Type] -> [Maybe String] -> [Exp]
 zipFields o = zipWithPred p (prettyField o)
   where
-    p :: Name -> Type -> Bool
-    p n _ = omitFields o (nameStr n) == Keep
+    p :: Name -> Type -> Maybe String -> Bool
+    p n _ _ = omitFields o (nameStr n) == Keep
 
-zipWithPred :: (a -> b -> Bool) -> (a -> b -> c) -> [a] -> [b] -> [c]
-zipWithPred _ _ [] _ = []
-zipWithPred _ _ _ [] = []
-zipWithPred p f (x : xs) (y : ys)
-  | p x y = f x y : zipWithPred p f xs ys
-  | otherwise = zipWithPred p f xs ys
+zipWithPred :: (a -> b -> c -> Bool) -> (a -> b -> c -> d) -> [a] -> [b] -> [c] -> [d]
+zipWithPred _ _ [] _ _ = []
+zipWithPred _ _ _ [] _ = []
+zipWithPred _ _ _ _ [] = []
+zipWithPred p f (x : xs) (y : ys) (z : zs)
+  | p x y z = f x y z : zipWithPred p f xs ys zs
+  | otherwise = zipWithPred p f xs ys zs
```

Comment:
hm, would probably prefer to see this as `catMaybes . zipWith3 f` where the `f` returns a `Maybe d` instead - but that's a *really* minor nit

Reply:
Doing it.